src/main/java/graphql/scalar/GraphqlIntCoercing.java

```diff
@@ -63,14 +63,46 @@ private Integer serialiseImpl(Object input, @NotNull Locale locale) {
     }
 
     @NotNull
-    private Integer parseValueImpl(Object input, @NotNull Locale locale) {
-        Integer result = convertImpl(input);
+    private Integer parseValueImpl(@NotNull Object input, @NotNull Locale locale) {
+        if (input instanceof Integer) {
+            return (Integer) input;
+        }
+
+        if (!(input instanceof Number)) {
+            throw new CoercingParseValueException(
+                    i18nMsg(locale, "Int.notInt", typeName(input))
+            );
+        }
+
+        BigInteger result = convertParseValueImpl(input);
         if (result == null) {
             throw new CoercingParseValueException(
                     i18nMsg(locale, "Int.notInt", typeName(input))
             );
         }
-        return result;
+
+        if (result.compareTo(INT_MIN) < 0 || result.compareTo(INT_MAX) > 0) {
+            throw new CoercingParseValueException(
+                    i18nMsg(locale, "Int.outsideRange", result.toString())
```

Comment:
For values outside of the integer range, we previously threw an exception with the same `Int.notInt` message. We ought to use `Int.outsideRange` for this case, as we do in `parseLiteral`